src/Form/DataTransformer/ModelToIdPropertyTransformer.php

```diff
@@ -108,12 +105,21 @@ public function reverseTransform($value)
             throw new \UnexpectedValueException(sprintf('Value should be array, %s given.', \gettype($value)));
         }
 
-        foreach ($value as $key => $id) {
-            if ('_labels' === $key) {
-                continue;
-            }
+        $ids = array_filter($value, static function ($key): bool {
+            return '_labels' !== $key;
+        }, ARRAY_FILTER_USE_KEY);
+
+        // NEXT_MAJOR: Remove this check and the `else` condition.
+        if (method_exists($this->modelManager, 'findMultiple')) {
+            /** @phpstan-var ArrayCollection<array-key, T> $collection */
```

Comment:
```suggestion
            /** @phpstan-var Collection<array-key, T> $collection */
```

And I assume that the comment won't be needed in next major, isn't it ?